File: src/main/java/com/kerimsenturk/labreport/dto/UserDto.java
package com.kerimsenturk.labreport.dto;

import com.kerimsenturk.labreport.model.enums.UserRole;

public class UserDto {
    String userId;
    String name;
    String surname;
    UserRole role;

    //We do not want to show secret information in JSON like password.
    //String password;

    public UserDto() {
    }

    public UserDto(String userId, String name, String surname, UserRole role) {
        this.userId = userId;
        this.name = name;
        this.surname = surname;
        this.role = role;
    }

    public String getUserId() {
        return userId;
    }

    public void setUserId(String userId) {
        this.userId = userId;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getSurname() {
        return surname;
    }

    public void setSurname(String surname) {
        this.surname = surname;
    }

    public UserRole getRole() {
        return role;
    }

    public void setRole(UserRole role) {
        this.role = role;
    }
}

File: src/main/web-ui/src/domain/dto/UserDto.js
"use strict";
exports.__esModule = true;
exports.UserDto = void 0;
var UserDto = /** @class */ (function () {
    function UserDto(userId, name, surname, role) {
        this._userId = userId;
        this._name = name;
        this._surname = surname;
        this._role = role;
    }
    Object.defineProperty(UserDto.prototype, "userId", {
        get: function () {
            return this._userId;
        },
        set: function (value) {
            this._userId = value;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(UserDto.prototype, "name", {
        get: function () {
            return this._name;
        },
        set: function (value) {
            this._name = value;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(UserDto.prototype, "surname", {
        get: function () {
            return this._surname;
        },
        set: function (value) {
            this._surname = value;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(UserDto.prototype, "role", {
        get: function () {
            return this._role;
        },
        set: function (value) {
            this._role = value;
        },
        enumerable: false,
        configurable: true
    });
    return UserDto;
}());
exports.UserDto = UserDto;
